kin-backup-and-restore/kin-backup-and-restore-ui-tests/src/androidTest/kotlin/kin/backup/and/restore/ui/tests/BackupActivityValidateSwitchBetweenScreens.kt

```diff
@@ -0,0 +1,263 @@
+    }
+
+    @Test
+    fun backupActivity_ThirdScreen_ToFourth_CorrectComponents() {
+        launchActivity()
+
+
+        val letsGoButton = onView(allOf(withId(R.id.lets_go_button), withText(R.string.backup_and_restore_lets_go)))
+        letsGoButton.perform(click())
+
+        val password = onView(
+                allOf(childAtPosition(
+                        allOf(withId(R.id.enter_pass_edittext),
+                                childAtPosition(
+                                        withClassName(`is`("android.support.constraint.ConstraintLayout")),
+                                        2)),
+                        0),
+                        isDisplayed()))
+        password.perform(replaceText("qwertyU1!"))
+
+        val confirm = onView(
+                allOf(childAtPosition(
+                        allOf(withId(R.id.confirm_pass_edittext),
+                                childAtPosition(
+                                        withClassName(`is`("android.support.constraint.ConstraintLayout")),
+                                        3)),
+                        0),
+                        isDisplayed()))
+        confirm.perform(replaceText("qwertyU1!"))
+
+        closeSoftKeyboard()
+
+        val understandCheckbox = onView(withId(R.id.understand_checkbox))
+        understandCheckbox.perform(click())
+
+        val nextButton = onView(withId(R.id.next_button))
+        nextButton.perform(click())
+
+        val toolbarTitle = onView(allOf(instanceOf(TextView::class.java), withParent(withId(R.id.toolbar))))
+        toolbarTitle.check(matches(withText(R.string.backup_and_restore_my_kin_wallet_qr_code)))
+
+        val button = onView(
+                allOf(withId(R.id.send_email_button), isEnabled(), isDisplayed()))
+        button.perform(click())
+
+        val device = UiDevice.getInstance(getInstrumentation())
+        device.pressBack()
+
+        val understandCheckbox2 = onView(withId(R.id.i_saved_my_qr_checkbox))
+        understandCheckbox2.perform(click())
+
+        val toolbarTitle2 = onView(allOf(instanceOf(TextView::class.java), withParent(withId(R.id.toolbar))))
+        toolbarTitle2.check(doesNotExist())
+
+        val textView = onView(withId(R.id.steps_text))
+        textView.check(matches(withText("")))
+                .check(matches(not(isDisplayed())))
+
+        val imageButton = onView(
+                allOf(childAtPosition(
+                        allOf(withId(R.id.toolbar),
+                                childAtPosition(
+                                        IsInstanceOf.instanceOf(android.view.ViewGroup::class.java),
+                                        0)),
+                        0),
+                        isDisplayed()))
+        imageButton.check(matches(isDisplayed()))
+                .check(matches(isSameDrawable(R.drawable.close)))
+
+        val f = Activity::class.java.getDeclaredField("mResultCode")
```

Comment:
🤔 do we have to do reflection here and leak impl details? can't we assert it by UI state?

Reply:
Didn't find a way to verify result...